perllib/FixMyStreet/App/Controller/Report.pm

```diff
@@ -660,6 +673,26 @@ sub map :Chained('id') :Args(0) {
     $c->res->body($image->{data});
 }
 
+sub confirmation :Chained('id') :Args(0) {
+    my ($self, $c) = @_;
+
+    my $template = $c->get_param('template') || "";
+
+    if ($template !~ /^(email_sent|confirm_problem)$/) {
+        $c->detach( '/page_error_404_not_found', [] );
+    }
+
+    if ($template eq 'email_sent') {
+        $c->stash->{template} = 'email_sent.html';
+        $c->stash->{email_type} = 'problem';
+    } elsif ($template eq 'confirm_problem') {
+        $c->stash->{template} = 'tokens/confirm_problem.html';
+        $c->stash->{created_report} = $c->get_param('created_report');
+        $c->stash->{report} = $c->stash->{problem};
```

Comment:
This lets me show a confirmation page for any report already in the system for any cobrand, e.g. this BANES report on the Bucks cobrand:
<img width="1076" alt="image" src="https://github.com/mysociety/fixmystreet/assets/154364/7635aa19-280f-424e-b656-27d18b5f06da">
Whilst I see your checks for non_public reports, this feels like something confusing/bad could potentially happen at some point (and I don't know all the combinations of `owns_problem` on this page when it's given any report at all that it might not be expecting). More generally, I'd just be worried about the possibility of something being made public by this that wasn't expected.

Reply:
I've implemented the token system you suggested so this should be OK now.